isis/src/control/objs/ControlNetVitals/ControlNetVitals.cpp

```diff
@@ -0,0 +1,245 @@
+    list.append("CASSIS_01.cub");
+    return list;
+  }
+
+  int ControlNetVitals::numPoints() {
+    return m_controlNet->GetNumPoints();
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numIgnoredPoints() {
+    int count = 0;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->IsIgnored()) count++;
+    }
+    return count;
+  }
+
+  int ControlNetVitals::numLockedPoints() {
+    return m_controlNet->GetNumEditLockPoints();
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numFixedPoints() {
+    int count = 0;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Fixed") count++;
+    }
+    return count;
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numConstraintedPoints() {
+    int count = 0;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Constrained") count++;
+    }
+    return count;
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numFreePoints() {
+    int count = 0;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Free") count++;
+    }
+    return count;
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numPointsBelowMeasureThreshold(int num) {
+    int count = 0;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetNumMeasures() < num) count++;
+    }
+    return count;
+  }
+
+  int ControlNetVitals::numImages() {
+    return m_controlNet->GetCubeSerials().size();
+  }
+
+  int ControlNetVitals::numMeasures() {
+    return m_controlNet->GetNumMeasures();
+  }
+
+  // REFACTOR
+  int ControlNetVitals::numImagesBelowMeasureThreshold(int num) {
+    int count = 0;
+    foreach(QString serial, m_controlNet->GetCubeSerials()) {
+      if (m_controlNet->GetMeasuresInCube(serial).size() < num) count++;
+    }
+    return count;
+  }
+  // REFACTOR
+  int ControlNetVitals::numImagesBelowHullTolerance(int tolerance) {
+    return 1;
+  }
+
+  QList<QString> ControlNetVitals::getCubeSerials() {
+    return m_controlNet->GetCubeSerials();
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getAllPoints() {
+    return m_controlNet->GetPoints();
+  }
+
+  // REFACTOR
+  QList<ControlPoint*> ControlNetVitals::getIgnoredPoints() {
+    QList<ControlPoint*> ignoredPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->IsIgnored()) ignoredPoints.append(point);
+    }
+    return ignoredPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getLockedPoints() {
+    QList<ControlPoint*> lockedPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->IsEditLocked()) lockedPoints.append(point);
+    }
+    return lockedPoints;
+  }
+
+  QList<ControlPoint*> ControlNetVitals::getFixedPoints() {
+    QList<ControlPoint*> fixedPoints;
+    foreach(ControlPoint* point, m_controlNet->GetPoints()) {
+      if (point->GetPointTypeString() == "Fixed") fixedPoints.append(point);
```

Comment:
Same as the counting methods, use the actual enumeration instead of string comparisons.